src/timing.rs

```diff
@@ -0,0 +1,183 @@
+use core::fmt::{Display, Formatter};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+pub enum Timing {
+    DnsLookup,
+    TcpConnection,
+    TlsHandshake,
+    ServerProcessing,
+    ContentTransfer,
+    HtmlParse,
+    CssParse,
+    JsParse,
+    RenderTree,
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct Timer {
```

Comment:
Timer is just a start and end time, and calculates the differences between (all in milliseconds)